x/mongo/driver/connstring/connstring.go

```diff
@@ -130,6 +130,14 @@ func (u *ConnString) String() string {
 	return u.Original
 }
 
+// HasAuthParameters returns true if this ConnString has any authentication parameters set and therefore represents
+// a request for authentication.
+func (u *ConnString) HasAuthParameters() bool {
```

Comment:
I added this function because it seemed like a leaky abstraction to have `ClientOptions` be aware of these details. It only needs to know if the ConnString represents a request for auth and to create a `Credential` if it does.

Reply:
Should we go further and just have a (nillable) Credential property for this struct?

Reply:
Without a more extensive refactor, this would create an import cycle. `ClientOptions` stores an `options.Credential` field, which can also be set by `ClientOptions.SetAuth`. The `Client` later translates `options.Credential` into `auth.Credential`, which has some extra information like the app name to use in the initial handshake.

Adding an `options.Credential` property to ConnString would create an import cycle between the `options` and `connstring` packages.

Reply:
Ack.  LGTM still.